### DISCUSSION TITLE
Stuck on terraform init - 'Requested entity already exists'

### DISCUSSION THREAD

**Discussion by kenzozozo** [GoogleCloudPlatform/fda-mystudies#3892] [Q&A]
I'm having trouble at the _terraform init_ step in the process.

Two issues:

1. It seems as though the process gets hung up irrevocably if it doesn't execute perfectly the first time?  If the deployment.hcl file and set_env_var.sh files are not absolutely set up correctly, and the IAM groups are not set up to the exact same names, then the _terraform apply_ step will fail, and an entirely new prefix/ENV will need to be entered in order to even attempt to run terraform again?  It doesn't seem to be able to recognize when google resources are soft deleted, and I keep getting an error telling me that the "entity {PREFIX}-{ENV}-dev/folder-admin" still exists.  Has this been an issue for anyone else?
2. I've been able to get further and further as I learn how to navigate the extremely touchy nature of Terraform, but could anyone confirm whether or not the IAM group names should or should not contain the domain?  The reason I ask it because I got the following error:  **irtpoc34-dev-folder-admins@domain.com does not exist., badRequest**.  I checked several times, and the group does indeed exist with the exact same spelling, but the domain is absent...  Because of the above issue, however, I have to re-do the entire deployment.hcl and set_env_var.sh files, and the groups again, otherwise _terraform apply_ will throw an error saying the entities already exist.

Any advice would be greatly appreciated.

Update: I was able to progress past this stage.  The group names must contain the domain.  e.g. **prefix-environment-folder-admins@domain.com**.  This seemed to be the major hang up.

Update 2: Seems the error resurfaces during cloud run.  Triggering cloud run by performing a merge in github successfully triggers the trigger, but errors out with the same "{myprefix}-{myenv}-devops already exists" error.  After hunting around, it appears this is a known issue within terraform itself.

Update 3: Was able to progress past this by starting entirely from scratch: Cloning the repo, resetting group names, resetting deployment.hcl and set_env_vars.sh with all of the necessary parameters.  Also deposited the requisite $10 to get my project count upped to 10 concurrent projects.  Cloud build throws an error on merge:

**"Apply": Error: error creating project poc9-dev-devops (poc9-dev-devops): googleapi: Error 409: Requested entity already exists, alreadyExists. If you received a 403 error, make sure you have the `roles/resourcemanager.projectCreator` permission**

Not sure how to proceed from here.

**Comment by itsvikrams** [comment]
I'm also facing the same error.Were you able to progress further?

**Reply by kenzozozo** [reply]
Yes!  

I suspect it's because the project specifies an older version of terraform, but I have no proof. Apparently, newer versions of terraform are able to delete/create/re-use projects more "intelligently," such that if the project already exists, it somehow knows to either re-use it or recreate it.

My solution was to delete all GCP projects and do everything exactly perfectly according to the directions **in one shot**. This means:

- deleting all projects (including removal of liens)
- re-naming security groups & config file appropriately
- deleting the entire repo and cloning it again
- following the setup instructions exactly, perfectly, in exact order

By doing this, I was able to progress to the cloud-build/git merge step, but cloud-build throws another error on merge. 

If you progress past that step, please share.

**Reply by Akebed** [reply]
I'm also facing the same issue. were you able to fix it?
